integreat_cms/static/src/js/language-tabs.ts

```diff
@@ -0,0 +1,96 @@
+const renderLanguageTabs = () => {
+    const globeElement = document.getElementById("language-switcher-globe");
+    const textElement = document.getElementById("language-switcher-text");
+    const tabWrapper = document.getElementById("tab-wrapper");
+    const languageSwitcher = document.getElementById("language-switcher");
+    const languageSwitcherListElement = document.getElementById("language-switcher").parentElement;
+    const languageSwitcherList = document.getElementById("language-switcher-list");
+
+    globeElement?.classList.add("hidden");
+    textElement?.classList.remove("hidden");
+
+    if (tabWrapper) {
+        const buffer = 20;
+        const paddingInWrapper =
+            parseFloat(window.getComputedStyle(tabWrapper).paddingLeft) +
+            parseFloat(window.getComputedStyle(tabWrapper).paddingRight);
+        const availableWidth = tabWrapper.offsetWidth - paddingInWrapper - languageSwitcherListElement.offsetWidth;
+        const tabs = Array.from(tabWrapper.children).concat(Array.from(languageSwitcherList.children));
+
+        let usedSpace = 0;
+        let tabCount = 0;
+
+        tabs.forEach((tab) => {
+            const marginOfTab = parseFloat(window.getComputedStyle(tab).marginRight);
+            const anchor = tab.getElementsByTagName("a")[0];
+
+            if (
+                usedSpace + tab.clientWidth + marginOfTab + buffer > availableWidth &&
+                tab !== languageSwitcherListElement &&
+                tabCount > 0
```

Comment:
It's difficult to visualize what's going on here, maybe pulling out the width calculation with an explicit name can make it easier. Also, I'd advocate for using [offsetWidth](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetWidth) over [clientWidth](https://developer.mozilla.org/en-US/docs/Web/API/Element/clientWidth), since it includes the border *(and potentially scroll bar, though not the margin)*.

```suggestion
        tabs.forEach((tab: HTMLElement) => {
            const marginOfTab = parseFloat(window.getComputedStyle(tab).marginRight);
            const anchor = tab.getElementsByTagName("a")[0];
            const fitsWithLanguageSwitcher = usedSpace + tab.offsetWidth + marginOfTab + buffer <= availableWidth;

            if (
                !fitsWithLanguageSwitcher &&
                tab !== languageSwitcherListElement &&
                tabCount > 0
```